### PR TITLE
Add job image to copy images from DockerHub to GCR

### PR DESCRIPTION
**Pull request by robinschneider**
/kind enhancement

**What this PR does / why we need it**:
Add job image to copy images from DockerHub to GCR

**Which issue(s) this PR fixes**:
Part of #619 

**Special notes for your reviewer**:

### CODE REVIEW

File: images/copy-images/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ARG ALPINE_VERSION=3.17
```
Comment (timebertt) [inline images/copy-images/Dockerfile:1]: Can you pin this to the latest patch version?

```suggestion
ARG ALPINE_VERSION=3.17.2
```

File: images/copy-images/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM alpine:$ALPINE_VERSION
+LABEL ALPINE_VERSION=$ALPINE_VERSION
+ARG CRANE_VERSION=v0.14.0
+LABEL CRANE_VERSION=$CRANE_VERSION
+RUN set -eux; \
+    apk update; \
```
Comment (timebertt) [inline images/copy-images/Dockerfile:8]: `apk update` should not be required when using `apk add --no-cache`

File: images/copy-images/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    apk add --no-cache yq \
+    curl; \
+    curl -L "https://github.com/google/go-containerregistry/releases/download/$CRANE_VERSION/go-containerregistry_Linux_amd64.tar.gz" -O; \
+    tar -xf go-containerregistry_Linux_arm64.tar.gz -C /usr/local/bin; \
```
Comment (timebertt) [inline images/copy-images/Dockerfile:12]: Can you make sure to remove the downloaded tarball from the resulting image?

File: images/copy-images/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+ARG ALPINE_VERSION=3.17
+
+FROM alpine:$ALPINE_VERSION
+LABEL ALPINE_VERSION=$ALPINE_VERSION
+ARG CRANE_VERSION=v0.14.0
+LABEL CRANE_VERSION=$CRANE_VERSION
+RUN set -eux; \
+    apk update; \
+    apk add --no-cache yq \
+    curl; \
```
Comment (timebertt) [inline images/copy-images/Dockerfile:10]: This is unneeded.

**Comment by robinschneider** [comment]
@timebertt Thanks for your remarks. I updated it and fixed an architecture mismatch I found.

**Review by timebertt — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**